packages/core/src/StateNode.ts

```diff
@@ -991,13 +989,6 @@ class StateNode<
       }
     }
 
-    if (!transition.source) {
-      transition.exitSet = [];
-
-      // Ensure that root StateNode (machine) is entered
-      transition.entrySet.push(this);
-    }
```

Comment:
this was redundant as in this scenario (when there is no `transition.source` - which in turn means "when resolving the initial state") because the `exitSet` is already empty and nothing pushes to it AND `entrySet` is already resolved with `this` in it - so we were always adding a duplicate root to the list here